src/main/java/no/itera/bloggingplatform/model/Peristable.java

```diff
@@ -0,0 +1,9 @@
+package no.itera.bloggingplatform.model;
+
+public interface Peristable {
+
+    Long getKey();
+
+    void setKey(Long id);
+
```

Comment:
How about adding `boolean exists()`?
We'd be able to remove checks like these from service layer:
```
if (category.getKey() != null) {
            return category;
}
```
That'd make it a bit cleaner.
I'd expect the repo/persistence layer to deal with it so I propose we move the check to `DefaultInMemoryRepository#create` : 
```
    @Override
    public T create(T toPersist) {
        if (toPersist.exists()) {
            return toPersist;
        }
        toPersist.setKey(SimpleIdGenerator.generate());
        inMemoryCache.put(toPersist.getKey(), toPersist);
        return toPersist;
    }
```

Reply:
I agree with the `exists()` method, but I would slightly adjust contract. If objects exist, then it should not try to return the same object, but rather throw an exception

Reply:
Yup, even better